pkg/ingester/ingester.go

```diff
@@ -2214,7 +2212,9 @@ func (i *Ingester) createTSDB(_ context.Context, userID string, walReplayConcurr
 	// this will actually create the blocks. If there is no data (empty TSDB), this is a no-op, although
 	// local blocks compaction may still take place if configured.
 	level.Info(userLogger).Log("msg", "Running compaction after WAL replay")
-	err = db.Compact()
+	// Note that we want to let TSDB creation finish without being interrupted by eventual context cancellation,
+	// so passing an independent context here
+	err = db.Compact(context.Background())
```

Comment:
@pracucci made me aware that chances are it's a bad idea to pass a request-tied context to TSDB creation, since we don't want to interrupt the latter if the request gets canceled. Thinking the whole thing through, I suggest we don't pass down context to `createTSDB` at all in this PR (the context argument to `createTSDB` was previously unused). If there are cases for which we _want_ to let context cancellation interrupt TSDB creation (during service startup), it's probably better to enable it in a separate PR?

Reply:
@colega my thinking too.

Reply:
Agree with Oleg.